lib/analytics.js

```diff
@@ -4,6 +4,16 @@ const BATCH_SIZE = process.env.HAPI_GAPI_BATCH_SIZE || 20
 const MAX_QUEUE_TIME = 14400000
 const debug = require('debug')('hapi-gapi')
 const ipAnonymise = require('ip-anonymize')
+const Joi = require('@hapi/joi')
+
+const attributionSchema = Joi.object({
```

Comment:
Suggest adding `.trim()` and enforcing a minimum length of 1 character to the required fields.

According to https://ga-dev-tools.appspot.com/campaign-url-builder/ source and medium are also required fields so these should also be marked with `.required()`

Reply:
Completely agree on trim, but require already enforces a minimum string length of 1, so isn't necessary.

I've already included source, via the 'or' designation at the end (`.or('source', 'id')`). Medium didn't seem to be a requirement according to my tests: in the Google analytics interface for GAFL non-prod, if you select localhost, I'd sent a curl requests that specified campaign name as 'campaign 10' and 'campaign 11' which also specified a source (for campaign 10) or an id (for campaign 11) but no other attribution values and these seemed to register.